zerver/tests/test_auth_backends.py

```diff
@@ -1039,32 +1047,69 @@ def stage_two_of_registration(self, result: HttpResponse, realm: Realm, subdomai
                                  parsed_url.path)
         self.assertTrue(uri.startswith('http://zulip.testserver/accounts/login/subdomain/'))
 
-        result = self.client_get(result.url)
+        if not user_avatar_url:
+            result = self.client_get(result.url)
+        else:
+            data['user_avatar_url'] = user_avatar_url
```

Comment:
shouldn't `data['user_avatar_url']` have the URL anyway? The auth process puts the avatar url in the dict in `redirect_and_log_into_subdomain`, so it should be here I think. So then we should be doing `self.assertEqual(data['user_avatar_url'], user_avatar_url)` here if that works correctly

Reply:
I was letting the `user_avatar_url` on that dict but when I ran all tests in `test_auth_backends.py`, one of the `test_github_oauth2` broke because the `user_avatar_url` was None as it was not passing the URL. So, I was a bit afraid to change another test as it was testing something else, other than github_avatar_url.

Reply:
But should I change the other test?